OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -287,6 +287,16 @@ public bool IsReachableTarget(Target target, bool allowMove)
 				&& (target.IsInRange(self.CenterPosition, GetMaximumRange()) || (allowMove && self.Info.HasTraitInfo<IMoveInfo>()));
 		}
 
+		public bool UnforcedAttackCanOnlyTargetEnemies()
+		{
+			// PERF: Avoid LINQ.
+			foreach (var armament in Armaments)
+				if (!armament.IsTraitDisabled && armament.Info.TargetStances > Stance.Enemy)
```

Comment:
You're right. However I did want to catch `None` here too. I've realised I can rework this a bit to achieve both.